### PR TITLE
Fix #2875 : Check local NameRegistry before querying the remote

### LINKED ISSUE #2875: A node should look up names locally before requesting the registry
We are currently hammering the registry with requests:
```
2022-01-18 08:50:16,990 Trace [agora.node.Registry] - Successfull GET /validator: boa1xqvalc7v34kr9crh4e882zmguvt3dgmtdhxtqx0wsljej5f9xdxl6xftcay => { type: TYPE.CNAME, payload: { data: { public_key: boa1xqvalc7v34kr9crh4e882zmguvt3dgmtdhxtqx0wsljej5f9xdxl6xftcay, addresses: [agora://eu-004.bosagora.io/], seq: 1642402944, ttl: 600 }, signature: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 }, utxo: 0x3245a62b3c4630a35ba999087866cf8e70ec600059dd1f2ec1760f86f4128c8b5d521734a65069961eb7c5547557173887399aee0bf98b1fa8b3386c1c6d81d6 }
2022-01-18 08:50:16,991 Trace [agora.node.Registry] - Successfull GET /validator: boa1xzval8mq887lkjsqwyl58xyrkxxz8mphv5dx9qv2z750fxvcs9gtvpal0dm => { type: TYPE.CNAME, payload: { data: { public_key: boa1xzval8mq887lkjsqwyl58xyrkxxz8mphv5dx9qv2z750fxvcs9gtvpal0dm, addresses: [agora://eu-003.bosagora.io/], seq: 1642402944, ttl: 600 }, signature: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 }, utxo: 0x76cc80093c546ec3132e73bdf9a7c71dc4af45ca6dd269a7af6d2e8ea6fe5f288f5f5e8ee1e8df7a42eb978b79201f341ee5082a4a8ee8ae35df7af07f530893 }
2022-01-18 08:50:16,992 Trace [agora.node.Registry] - Successfull GET /validator: boa1xzval2a3cdxv28n6slr62wlczslk3juvk7cu05qt3z55ty2rlfqfc6egsh2 => { type: TYPE.CNAME, payload: { data: { public_key: boa1xzval2a3cdxv28n6slr62wlczslk3juvk7cu05qt3z55ty2rlfqfc6egsh2, addresses: [agora://eu-002.bosagora.io/], seq: 1642402544, ttl: 600 }, signature: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 }, utxo: 0xa3c2660bd4c65d6153bbe5bfac2cff06f68364c92e1f2619a4faede13943267df958efd9117c87125180241d678ff7278557fb8e83e486872a1eb8d83c41b4e1 }
2022-01-18 08:50:16,993 Trace [agora.node.Registry] - Successfull GET /validator: boa1xrval5rzmma29zh4aqgv3mvcarhwa0w8rgthy3l9vaj3fywf9894ycmjkm8 => { type: TYPE.CNAME, payload: { data: { public_key: boa1xrval5rzmma29zh4aqgv3mvcarhwa0w8rgthy3l9vaj3fywf9894ycmjkm8, addresses: [agora://na-001.bosagora.io:3826/], seq: 1642402547, ttl: 600 }, signature: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 }, utxo: 0x0666c4d505b55b6840fbb669ec08a1849e699d5a30ba246989b65ea71292f8ac9a3d7126ca9061313d3225d6e324146f37cdc5dab51facbbc3beead6854e89a4 }
2022-01-18 08:50:16,994 Trace [agora.node.Registry] - Successfull GET /validator: boa1xrval7gwhjz4k9raqukcnv2n4rl4fxt74m2y9eay6l5mqdf4gntnzhhscrh => { type: TYPE.CNAME, payload: { data: { public_key: boa1xrval7gwhjz4k9raqukcnv2n4rl4fxt74m2y9eay6l5mqdf4gntnzhhscrh, addresses: [agora://na-002.bosagora.io:3826/], seq: 1642402549, ttl: 600 }, signature: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 }, utxo: 0x6bceb7f5997df362bd0808826ef9577d7f8389bf9c75a471362aa5a24f1d64c9074850f29c6c078e2a7fa8555ec81fde75c2c8b6a8c0adc8d75057bbcb51116a }
2022-01-18 08:50:16,995 Trace [agora.node.Registry] - Successfull GET /validator: boa1xzval3ah8z7ewhuzx6mywveyr79f24w49rdypwgurhjkr8z2ke2mycftv9n => { type: TYPE.CNAME, payload: { data: { public_key: boa1xzval3ah8z7ewhuzx6mywveyr79f24w49rdypwgurhjkr8z2ke2mycftv9n, addresses: [agora://eu-002.bosagora.io:3826/], seq: 1642402545, ttl: 600 }, signature: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 }, utxo: 0x2b1fcfb62b868a53287fbdfbc17631f5a4e1cfdb91e58dfc0a8ebf083e92994dd327b0b574c81f0e29394d3abbf67f609ad67bce49c9d9b59672fa5a61a37495 }
2022-01-18 08:50:16,996 Trace [agora.node.Registry] - Successfull GET /validator: boa1xpval9gv8wjk5s05w0vplpgd5wrmzlvhj4e6zym302f2t6xeklzw2meepv9 => { type: TYPE.CNAME, payload: { data: { public_key: boa1xpval9gv8wjk5s05w0vplpgd5wrmzlvhj4e6zym302f2t6xeklzw2meepv9, addresses: [agora://eu-003.bosagora.io:3826/], seq: 1642402946, ttl: 600 }, signature: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 }, utxo: 0x4b92b5792c3bac487d7603bb33ceb07b7988dc057fea236f0041623cd472013cfb3ff5fcfe01cbaac24a7bdc6dab1b6882d6c026ca72767a92e7d6bb0c028753 }
2022-01-18 08:50:16,997 Trace [agora.node.Registry] - Successfull GET /validator: boa1xrval6hd8szdektyz69fnqjwqfejhu4rvrpwlahh9rhaazzpvs5g6lh34l5 => { type: TYPE.CNAME, payload: { data: { public_key: boa1xrval6hd8szdektyz69fnqjwqfejhu4rvrpwlahh9rhaazzpvs5g6lh34l5, addresses: [agora://na-002.bosagora.io/], seq: 1642402545, ttl: 600 }, signature: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 }, utxo: 0x84daa1d361afd9238041431b838500103eeb16d7952486077830a31a77ecc31d40f3a9d3f3a1c92dc3a0e5723ba6de9d87f107a73c73aef6a45bf9862c663493 }
2022-01-18 08:50:16,998 Trace [agora.node.Registry] - Successfull GET /validator: boa1xrvaldd5au5d5xs6pd6js7zah6m4h5d0r5tpwjasp99gvz3gmj2ex432u5x => { type: TYPE.CNAME, payload: { data: { public_key: boa1xrvaldd5au5d5xs6pd6js7zah6m4h5d0r5tpwjasp99gvz3gmj2ex432u5x, addresses: [agora://eu-004.bosagora.io:3826/], seq: 1642402944, ttl: 600 }, signature: 0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 }, utxo: 0x1996eb7e94e1483345809f9fb7251bed47951ac2997b23ee6e0bbeb9f4517520829ec331fd82457ba6b6f5c14d22ce2b9ec8af91f18e550759ff356871af41da }
[...]
```

Because we have absolutely no caching in place. Once #2703 is merged, we should look up locally first, instead of going straight to the registry.

### PR DESCRIPTION
**Pull request by Geod24**
Only for validator, as for Flash, we don't have `KnownChannel` yet.
Will need to add test, and probably to move it somewhere else so it behaves the same in the unittests.

### CODE REVIEW

**Comment by omerfirmak** [comment]
is this a stop gap until the caching registry is merged?

File: source/agora/network/Manager.d [hunk 1]
```diff
@@ -1243,7 +1249,60 @@ public class NetworkManager
         settings.httpClientSettings.readTimeout = timeout;
         settings.httpClientSettings.proxyURL = this.proxy_url;
 
-        return new RestInterfaceClient!NameRegistryAPI(settings);
+        auto remote =  new RestInterfaceClient!NameRegistryAPI(settings);
```
Comment (omerfirmak) [inline source/agora/network/Manager.d:1093]: What if we dont have a `remote` but only a local one? L1244 would return early.

Comment (mkykadir) [reply source/agora/network/Manager.d:1093]: This would be a welcome change after mentioned issue gets resolved. Even it won't be problem after #2703

**Comment by omerfirmak** [comment]
Rebased

**Comment by Geod24** [comment]
I think this can be simplified now.

**Comment by omerfirmak** [comment]
and it immediately has conflicts again :D

**Comment by Geod24** [comment]
So this won't work quite well anymore because the `Registry` depends on the `Network` (as it should), so having the `Network` depends on the registry is... Meh. I'll try to find a solution that breaks the circular dependency.

**Comment by Geod24** [comment]
Will take another approach.

**Pull request closed without merging**